packages/lesswrong/lib/collections/posts/collection.ts

```diff
@@ -8,6 +8,7 @@ import { userIsPostGroupOrganizer } from './helpers';
 const options: MutationOptions<DbPost> = {
   newCheck: (user: DbUser|null) => {
     if (!user) return false;
+    if (user.deleted) return false;
     return userCanDo(user, 'posts.new')
```

Comment:
This seems like the wrong place to enforce this. Isn't there a way in the user <-> posts.new relation that can run logic?

Reply:
Yeah, I don't actually really like the `userCanDo` paradigm, and would have preferred collections to generally just have generic permission functions. On the margin, I prefer doing things in readable code, as opposed to having permissions like this handled in some indirect way on the Collection object.